### PR TITLE
Added multi-profile support

### PR DESCRIPTION
**Pull request by PhoenixPeca**
Dear caddy team, this Pull-Request goes along with my other PR at the 'caddy-dns' organization: PR https://github.com/libdns/route53/pull/1..

> Dear caddy team, this PR adds a new argument in the route53 module where people can specify which AWS profile they want to use per virtualhost.. If not defined, it will fall back to the default profile or the profile set in the system environment..
>
> ---
> 
> Example:
>
> ```
>   tls {
>       dns route53 {
>         max_retries 10
>         aws_profile "production-profile" # New argument for multi-profile support
>       }
>   }
> ```

For details please read this forum thread: https://caddy.community/t/automatic-https-for-route53-multi-profile-support/10512/8

### CODE REVIEW

File: go.mod
Comment (mholt) [inline go.mod]: Why the downgrade?

File: go.mod
Comment (mholt) [inline go.mod]: Why using an untagged version? Anything needed in this commit?

File: provider.go
Comment (mholt) [inline provider.go:14]: Prefer `AWSProfile` since AWS is an initialism.

File: client.go
Comment (mholt) [inline client.go:25]: Region? Where?

File: client.go
Comment (mholt) [inline client.go:30]: Why?

File: client.go
Comment (mholt) [inline client.go:22]: This is probably obvious

**Review by mholt — CHANGES_REQUESTED** [review]
I don't use Route53 so this will need a review by @danlsgiga before it gets merged.

**Comment by danlsgiga** [comment]
Have you tested this using EC2 Instance profiles to see how it behaves (not providing any profile)?

File: client.go
Comment (PhoenixPeca) [reply client.go:25]: Hi, I think we can leave out aws-region.. Route53 doesnt require region since it is available globally. 

Screen Shot 2020-11-24 at 7 27 27 AM

---

eitherway, should there be a need for us to specify a region (which is unlikely), it should be done in the `~/.aws/config` file, not directly on the code.

File: client.go
Comment (PhoenixPeca) [reply client.go:30]: perhaps @danlsgiga can help answer.

Comment (danlsgiga) [reply client.go:30]: @mholt, I believe this is required to enable multiple profiles to use the same `~/.aws/config` file among all specified profiles... otherwise, it requires a specific block in the config file for each profile.

Comment (mholt) [reply client.go:30]: Gotcha. Ok, either the comment should explain that or we can remove it if it's obvious to people familiar with the library/service.

Comment (danlsgiga) [reply client.go:30]: Kewl... @PhoenixPeca can you please remove the comment? I think this is redundant and I agree with Matt here!

Once that is done, it LGTM and you can merge it @mholt! ;)

File: client.go
Comment (danlsgiga) [reply client.go:25]: Same thing here, please remove the comment.

File: client.go
Comment (PhoenixPeca) [reply client.go:30]: Hi @danlsgiga , sorry for late reply. I will edit now, but i havent had the time to test this with your recent question about EC2 instance profiles. Will do my best to test as soon as possible.

Comment (PhoenixPeca) [reply client.go]: > Have you tested this using EC2 Instance profiles to see how it behaves (not providing any profile)?

Hi @danlsgiga , so far working fine on my side. :D

**Comment by PhoenixPeca** [comment]
> Have you tested this using EC2 Instance profiles to see how it behaves (not providing any profile)?

can confirm this working :D

**Comment by francislavoie** [comment]
I'm a bit confused - https://github.com/caddy-dns/route53/pull/5 was merged before this PR was merged. That's an issue, because now `master` of the `caddy-dns` repo is using a struct field that doesn't exist in `master` on this repo.

**Comment by PhoenixPeca** [comment]
Hi @francislavoie , i forked the two repos last november and tested build.. Everything works for me, i think this is good for merge.. But, Ill leave the decisions to the maintainer, since they seem to have no time in checking this soon..

Right now, Im just using these two repos:

- https://github.com/PhoenixPeca/route53-1
- https://github.com/PhoenixPeca/route53

I use them in production. Basically just more or less a copy of the original repo + my PRs..
Can no longer wait any longer for this feature to be merged here, so I figured to just setup my own repo for my use case.. 😅😅😅

Looking forward to a release with this PR.. \o/

Happy New Year to all! ;)

**Comment by francislavoie** [comment]
Okay - @danlsgiga or @mholt please merge 😁

**Comment by danlsgiga** [comment]
LGTM! Thanks for working on this @PhoenixPeca and sorry for the delay!

**Pull request merged**
